### PR TITLE
Bits and pieces

### PR DESCRIPTION
**Pull request by nosnilmot**
Miscellaneous fixes/tweaks:

- Add `--with-schannel` to autotools `./configure` options
- Fix some docs and tests when builddir != srcdir
- Fix running tests with OpenSSL 3
- Fixes for PKCS12 support on Windows
- Hide coding style changes from `git blame`

### CODE REVIEW

**Comment by nosnilmot** [comment]
I think the LibreSSL/valgrind CI test failure is a false positive, or at least a problem in LibreSSL and not in the changes in this PR - it is triggered by the new pkcs12 file added in [cceaa8c](https://github.com/strophe/libstrophe/pull/208/commits/cceaa8c58ad5201a925830e0eb78c84389e2dfa4) and appears to be caused by something deep inside `PKCS12_parse()`

**Comment by sjaeckel** [comment]
https://github.com/libressl-portable/portable/issues/760

**Comment by sjaeckel** [comment]
Is there a specific reason for 8efaabc?

I would prefer to not increase our code complexity "just because it's possible to do something"

also: could you please rebase on top of master?

**Comment by nosnilmot** [comment]
> Is there a specific reason for [8efaabc](https://github.com/strophe/libstrophe/commit/8efaabc95615d4b9933ed79cfaa999593a90385b)?

Short answer: just because

Longer answer:
I think it's more common where separate cert and key options exist but it is possible to include both in the same file, for the combined file to be specified in the cert option (cf. Apache httpd - `SSLCertificateFile`, nginx - `ssl_certificate`, curl `--cert`, etc.) instead of the key option.
I originally implemented this as a change to require use of cert for PKCS#12 files instead of key, but failed to submit the PR before libstrophe 0.12.0 was released, and it would break API/ABI to do that now.
Recently I realised it was possible to accept it in either key or cert without a huge increase in complexity, so I did this.

> I would prefer to not increase our code complexity "just because it's possible to do something"

ok, if you want me to drop this I will.

> also: could you please rebase on top of master?

done.

**Comment by sjaeckel** [comment]
> I think it's more common where separate cert and key options exist but it is possible to include both in the same file, for the combined file to be specified in the cert option

That's an ooopsie then on my side... seems like I didn't do my research properly... is there some option that behaves differently in those tools when only a key is given?

**Comment by nosnilmot** [comment]
> That's an ooopsie then on my side... seems like I didn't do my research properly... is there some option that behaves differently in those tools when only a key is given?

I'm not aware of any use case in the apps/tools I am familiar with where only the key parameter would be specified.

File: src/conn.c [hunk 1]
```diff
@@ -595,7 +595,10 @@ void xmpp_conn_set_client_cert(xmpp_conn_t *const conn,
         conn->tls_client_cert = NULL;
     if (conn->tls_client_key)
         strophe_free(conn->ctx, conn->tls_client_key);
-    conn->tls_client_key = strophe_strdup(conn->ctx, key);
+    if (key)
+        conn->tls_client_key = strophe_strdup(conn->ctx, key);
+    else
+        conn->tls_client_key = NULL;
```
[the remark this answers is not in the document]
Comment (nosnilmot) [reply src/conn.c:601]: yes, that would be cleaner, although I would also suggest changing the other files to use tls_client_cert internally for the pkcs12 case just to be consistent with the convention I already mentioned

**Comment by sjaeckel** [comment]
> I originally implemented this as a change to require use of cert for PKCS#12 files instead of key, but failed to submit the PR before libstrophe 0.12.0 was released, and it would break API/ABI to do that now.

that would've been great :) we're going to have an ABI breakage scenario in the near future when we remove those API's that are marked as deprecated now. Maybe we should immediately deprecate the way of passing the P12 file via the key now? So people don't start using it in the first place. Printing a warning when someone uses it that way and making it clear from the documentation should be enough, or not!? 

I'm going to create a 0.12.1 as soon as this PR and the OpenSSL3 PR is merged.

File: src/conn.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (sjaeckel) [reply src/conn.c:601]: > although I would also suggest changing the other files to use tls_client_cert internally

definitely

**Comment by nosnilmot** [comment]
> Maybe we should immediately deprecate the way of passing the P12 file via the key now? So people don't start using it in the first place. Printing a warning when someone uses it that way and making it clear from the documentation should be enough, or not!?

That's what I did in [3d01915](https://github.com/strophe/libstrophe/pull/208/commits/3d0191540743865ace9a21bba014157394dec663), I guess that's the only way to find out if it is enough :)

**Comment by sjaeckel** [comment]
Thx a lot for this PR!

**Pull request merged**